src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final Circle defaultCircle = new Circle(10, Color.WHITE);
+
+    public Figure getRandomFigure() {
+        int randomFigureType = random.nextInt(5);
+        int randomSide = random.nextInt() * 16;
+        int randomMultiplier = random.nextInt(3) + 2;
+
+        switch (randomFigureType) {
+            case 0:
+                return new Square(randomSide, randomColor.getRandomColor());
+            case 1:
+                return new Rectangle(randomSide,
+                        randomSide * randomMultiplier, randomColor.getRandomColor());

```

Comment:
Please, pay attention [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor).